File: frontend/admin/src/models/Department.ts
import type { TeamDetail } from "./Teams";

export interface DepartmentList {
    id: number;
    name: string | null;
    headFirstName: string | null;
    headLastName: string | null;
    isVideoGreetingUploaded: boolean;
    teamCount: number;
}

export interface DepartmentDetail {
    id: number;
    name: string | null;
    headFirstName: string | null;
    headLastName: string | null;
    headVideoUrl: string | null;
    headMicrosoftTeamsUrl: string | null;
    teams: TeamDetail[];
}

export interface DepartmentLookup {
    id: number;
    name: string | null;
}

File: backend/BuddyBot/Domain/Entities/Department.cs
namespace Domain.Entities;
public class Department
{
    public int Id { get; init; }
    public string Name { get; set; }
    public string? HeadFirstName { get; set; }
    public string? HeadLastName { get; set; }
    public string? HeadVideoUrl { get; private set; }
    public string? HeadMicrosoftTeamsUrl { get; private set; }
    public bool IsDeleted { get; private set; }

    public List<Team> Teams { get; private set; }

    public Department( string name, string headFirstName, string headLastName )
    {
        Name = name;
        HeadFirstName = headFirstName;
        HeadLastName = headLastName;
        Teams = new List<Team>();
    }

    public void AddHeadVideo( string headVideoUrl )
    {
        HeadVideoUrl = headVideoUrl;
    }

    public void AddHeadMicrosoftTeamsLink( string headMicrosoftTeamsUrl )
    {
        if ( string.IsNullOrWhiteSpace( headMicrosoftTeamsUrl) )
            throw new ArgumentException( "URL не может быть пустым." );

        HeadMicrosoftTeamsUrl = headMicrosoftTeamsUrl;
    }

    public void SoftDelete()
    {
        IsDeleted = true;
    }
}
